src/main/java/sg/dex/starfish/Job.java

```diff
@@ -176,4 +176,37 @@ public default boolean cancel(boolean mayInterruptIfRunning) {
 		return false;
 	}
 
+
+	/**
+	 * Gets the status of the Job associated with this Operation.
+	 *
+	 * @return The the status of the Job
+	 */
+	Status getStatus();
+
+	/**
+	 * This enum is for different job status
+	 */
+	enum Status {
```

Comment:
Would prefer a simple String status I think, to be consistent with the other JSON-consuming aspects of Starfish and enable extensibility without breaking code.

Reply:
updated